### DISCUSSION TITLE
Uploading images from Image Picker in React Native Expo

### DISCUSSION THREAD

**Discussion by rodjoseph** [supabase/supabase#1268] [Questions] [answered]
In order to upload images to supabase storage, you need a 'File' object. In my application, I can have the user either take a picture from the camera or pick one from the system's image picker, but the result is returned as a file uri (string). I attempt to fetch the uri like so:

`const uploadImage = async() => {
    const response = await fetch(newCollectionItemImageUri)
    const blob = await response.blob();

    const { data, error } = await supabase
      .storage
      .from('itemimages')
      .upload('public/' , blob)
  }`

But I get the TypeScript error:
`Argument of type 'Blob' is not assignable to parameter of type 'File'.
  Type 'Blob' is missing the following properties from type 'File': lastModified, namets(2345)`

Does anyone know how I can construct a `File `object to be used for the API using React Native (specifically, I am using Expo)

**Comment by inian** [comment, marked as answer]
Hi @rodjoseph, can you try converting the Blob to a File object like [this](https://stackoverflow.com/a/27256755)?

**Reply by rodjoseph** [reply]
Looks like that will work. Thank you. Later on, I'll be explaining the whole process for future reference for people who might want to use Expo with Supabase

**Reply by inian** [reply]
That will be very helpful @rodjoseph! Ping me when you have something and I will add it to our [examples page](https://supabase.io/docs/guides/examples).

**Reply by rodjoseph** [reply]
I have a follow up question.... is there a way I can get a URL to reference public images from. In my database, the rows have image URLs, and I want to be able to use those to reference the uploaded image. Can `upload` or something similar used to access the file afterwards possibly return a URL? (Not a temporary signed URL)

**Reply by inian** [reply]
Right now there is no way to make an object public without creating a signed URL. We are planning to allow users to make buckets fully public in the future. In the meantime you can create signedURLs with a very long expiry time (like 10 years for ex)

**Reply by neerajdana996** [reply]
const { data, error } = await supabase
                .storage
                .from('pics')

                .upload(fileName, new File([blob],fileName))
                 console.log(data, error);
                 
                 
 This give error

**Reply by PrestoAnomaly** [reply]
> Looks like that will work. Thank you. Later on, I'll be explaining the whole process for future reference for people who might want to use Expo with Supabase

@rodjoseph Did you ever happen to share an example? Having similar issues with Expo + Image Picker. Thanks!

**Reply by thareekanvar** [reply]
@PrestoAnomaly 
```
 const body = new FormData();

    body.append('upload', {
      uri: result,
      name: res[0].name,
      type: res[0].type,
    });
```
try this
